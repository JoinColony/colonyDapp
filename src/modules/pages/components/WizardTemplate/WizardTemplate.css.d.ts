export const query700: string;
export const layoutMain: string;
export const header: string;
export const logo: string;
export const content: string;
export const section: string;
export const logoLoader: string;
export const stepBarContainer: string;
