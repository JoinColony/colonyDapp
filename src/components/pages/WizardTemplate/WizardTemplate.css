.layoutMain {
  composes: stretchVertical stretchHorizontal from '~styles/layout.css';
  overflow: auto;
  background-color: var(--grey-blue-2);
}

.header {
  display: flex;
  justify-content: space-between;
  padding-top: 8px;
  height: 60px;
  width: 100%;
  background-color: white;
}

.logo {
  flex: 1;
  margin-left: 13px;
  max-width: 90px;
}

.content {
  composes: flexContainerColumn flexAlignCenter flexJustifyCenter from '~styles/layout.css';
  padding-bottom: 20px;
}

.section .logo-loader svg {
  max-width: 40%;
}

.stepBarContainer {
  padding-right: 20px;
}
