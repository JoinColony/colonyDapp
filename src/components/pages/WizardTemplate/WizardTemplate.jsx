/* @flow */
import type { Node } from 'react';

import React from 'react';
import { Link } from 'react-router-dom';

import Logo from '../../../img/logo.svg';
import { StepBar } from '~components/core/ProgressBar';
import styles from './WizardTemplate.css';

type Props = {|
  children: Node,
  step?: number,
  stepCount?: number,
|};

const displayName = 'pages.WizardTemplate';

const WizardTemplate = ({ children, step, stepCount }: Props) => (
  <main className={styles.layoutMain}>
    <header className={styles.header}>
      <figure className={styles.logo} role="presentation">
        <Link to="/">
          <Logo />
        </Link>
      </figure>
      {stepCount && step && (
        <div className={styles.stepBarContainer}>
          <StepBar step={step} stepCount={stepCount} />
        </div>
      )}
    </header>
    <article className={styles.content}>{children}</article>
  </main>
);

WizardTemplate.displayName = displayName;

export default WizardTemplate;
