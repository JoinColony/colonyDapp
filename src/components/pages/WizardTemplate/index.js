/* @flow */

export { default } from './WizardTemplate.jsx';
