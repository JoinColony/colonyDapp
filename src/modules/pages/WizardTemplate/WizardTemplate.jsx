/* @flow */

import React from 'react';
import type { Node } from 'react';

import Logo from '../../../img/logo.svg';
import { StepBar } from '../../core/components/ProgressBar';
import styles from './WizardTemplate.css';

type Props = {
  children: Node,
  step?: number,
  stepCount?: number,
  internal: boolean,
};

const displayName = 'pages.WizardTemplate';

const WizardTemplate = ({ children, step, stepCount, internal }: Props) => (
  <main className={styles.layoutMain}>
    <header className={styles.header}>
      <figure className={styles.logo} role="presentation">
        <Logo />
      </figure>
      {stepCount &&
        step && (
          <div className={styles.stepBarContainer}>
            <StepBar step={step} stepCount={stepCount} />
          </div>
        )}
    </header>
    <article className={styles.content}>{children}</article>
  </main>
);

WizardTemplate.displayName = displayName;

export default WizardTemplate;
