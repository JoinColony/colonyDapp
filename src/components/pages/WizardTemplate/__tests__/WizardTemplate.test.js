/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from 'testutils';

import WizardTemplate from '../WizardTemplate.jsx';

describe('WizardTemplate component', () => {
  test('Renders initial component', () => {
    const component = shallowWithIntl(
      <WizardTemplate stepCount={2} step={1} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
