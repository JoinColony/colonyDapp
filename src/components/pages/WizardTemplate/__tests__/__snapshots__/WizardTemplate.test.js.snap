// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardTemplate component Renders initial component 1`] = `
<main>
  <header>
    <figure
      role="presentation"
    >
      <Link
        replace={false}
        to="/"
      >
        <test-file-stub />
      </Link>
    </figure>
    <div>
      <InjectIntl(StepBar)
        step={1}
        stepCount={2}
      />
    </div>
  </header>
  <article />
</main>
`;
